src/fswAlgorithms/attDetermination/inertialAttitudeUkf/inertialAttitudeUkf.cpp

```diff
@@ -0,0 +1,303 @@
+        if (starTracker.timeTag*NANO2SEC > this->previousFilterTimeTag){
+            auto starTrackerMeasurement = Measurement();
+            starTrackerMeasurement.name = "starTracker";
+            starTrackerMeasurement.timeTag = starTracker.timeTag*NANO2SEC;
+            starTrackerMeasurement.validity = true;
+            starTrackerMeasurement.size = 3;
+
+            starTrackerMeasurement.noise = this->starTrackerMessages[index].measurementNoise;
+            starTrackerMeasurement.observation = mrpSwitch(Eigen::Map<Eigen::Vector3d>(starTracker.MRP_BdyInrtl),
+                    this->mrpSwitchThreshold);
+            starTrackerMeasurement.model = mrpFirstThreeStates;
+            this->measurements[this->measurementIndex] = starTrackerMeasurement;
+            this->measurementIndex += 1;
+            this->validStarTracker = true;
+            /*! - Only consider the filter started once a Star Tracker image is processed */
+            if (this->firstFilterPass){this->firstFilterPass = false;}
+        }
+        else{this->validStarTracker=false;}
+    }
+}
+
+/*! Loop through the entire gyro buffer to find the first index that is in the future compared to the
+* previousFilterTimeTag. This does not assume the data comes in chronological order since the gyro data
+* is a ring buffer and can wrap around
+* @return void
+* */
+void InertialAttitudeUkf::readGyroData(){
+        int smallestFutureIndex = 0;
+        int numberOfValidGyroMeasurements = 0;
+        double firstFutureTime = -1;
+        double meanMeasurementTime = 0;
+        AccDataMsgPayload gyrBuffer = this->accelDataMsg();
+        for (int index = 0; index < MAX_ACC_BUF_PKT; index++) {
+            double gyroMeasuredTime = gyrBuffer.accPkts[index].measTime*NANO2SEC;
+            if (gyroMeasuredTime > this->previousFilterTimeTag) {
+                if (gyroMeasuredTime < firstFutureTime || firstFutureTime<0){
+                    smallestFutureIndex = index;
+                    firstFutureTime = gyroMeasuredTime;
+                }
+                meanMeasurementTime += gyroMeasuredTime;
+                numberOfValidGyroMeasurements += 1;
+            }
+        }
+        auto lowPass = LowPassFilter();
+        lowPass.setFilterCutoff(this->cutOffFrequency);
+        lowPass.setFilterStep(this->hStep);
+        if (numberOfValidGyroMeasurements > 0){
+            meanMeasurementTime /= numberOfValidGyroMeasurements;
+            /*! - Loop through buffer for all future measurements since the previous time to filter omega_BN_B*/
+            for (int index = 0; index < MAX_ACC_BUF_PKT; index++) {
+                int shiftedIndex = (index + smallestFutureIndex) % MAX_ACC_BUF_PKT;
+                auto omega_BN_B = Eigen::Map<Eigen::Vector3d>(gyrBuffer.accPkts[shiftedIndex].gyro_B);
+                /*! - Apply low-pass filter to gyro measurements to get smoothed body rate*/
+                lowPass.processMeasurement(omega_BN_B);
+                }
```

Comment:
:large_blue_circle: This brace is over indented.

Reply:
Fixed